src/Database/Database.php

```
@@ -3663,6 +3730,18 @@ public function createDocument(string $collection, Document $document): Document
 
         $document = $this->encode($collection, $document);
 
+        // Debug: Check if spatial data is properly encoded
+        if ($this->adapter->getSupportForSpatialAttributes()) {
+            $attributes = $collection->getAttribute('attributes', []);
+            foreach ($attributes as $attribute) {
+                $key = $attribute['$id'] ?? '';
+                $type = $attribute['type'] ?? '';
+                if (in_array($type, [self::VAR_GEOMETRY, self::VAR_POINT, self::VAR_LINESTRING, self::VAR_POLYGON])) {
+                    $value = $document->getAttribute($key);
```

PHPMD (2.15.0)

Avoid unused local variables such as '$value'. (Unused Code Rules)

(UnusedLocalVariable)